File: Java/src/main/java/Solution801_900/Solution832.java
package Solution801_900;

/**
 * @author windows
 */
public class Solution832 {
    public int[][] flipAndInvertImage(int[][] image) {
        for (int i = 0; i < image.length; i++) {
            for (int j = 0, k = image[i].length - 1; j < k; j++, k--) {
                int temp = image[i][j];
                image[i][j] = image[i][k];
                image[i][k] = temp;
            }
            for (int j = 0; j < image[i].length; j++) {
                if (image[i][j] == 0){
                    image[i][j] = 1;
                }else{
                    image[i][j] = 0;
                }
            }
        }
        return image;
    }


    public static void main(String[] args) {
        Solution832 solution832 = new Solution832();
    }
}

File: C++/src/Solution801_900/Solution832.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution832.h"
#include <functional>
#include <algorithm>
#include <iostream>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <vector>
using namespace std;

vector<vector<int>> flipAndInvertImage(vector<vector<int>>& image) {
  for (auto &item : image){
    reverse(item.begin(),item.end());
  }
  for (auto &rol : image){
    for (auto &item : rol){
       item = item ^ 1;
    }
  }
  return image;
}
